nb_anacondacloud/static/main.js

```diff
@@ -385,17 +382,113 @@ function ($, dialog, Jupyter) {
         Jupyter.notification_area.get_widget('notebook').
             danger(title, 4000);
 
-        body = $('<div>').append(notif);
+        if (showLoginForm) {
+            promptLogin();
+        } else {
+            body = $('<div>').append(notif);
+
+            dialog.modal({
+                title: title,
+                body: body,
+                buttons : {
+                    OK: {}
+                }
+            });
+        }
+    }
+
+    function promptLogin(withError) {
+        var body,
+            form,
+            username,
+            password,
+            modal;
+        body = $('<div/>');
+
+        // avoid any surprises with user events firing in the background
+        Jupyter.notebook.keyboard_manager.register_events(body);
+        if (withError) {
+            $('<div/>', {'class': 'alert alert-danger', 'role': 'alert'})
+                .text('Invalid username or password')
+                .appendTo(body);
+        }
+
+        // warning about SSL
+        $('<div/>', {'class': 'alert alert-warning', 'role': 'alert'})
+            .html('If you are not under a secure connection we <b>strongly</b> recommend to ' +
+                  'login through the command line with: <pre>anaconda login</pre>')
+            .appendTo(body);
 
-        dialog.modal({
-            title: title,
+        // actually build the form
+        form = $('<div/>', {
+            'class': 'form-horizontal',
+            'role': 'form'
+        }).appendTo(body);
+
+        username = $('<input/>', {
+            'class': 'form-control',
+            type: 'text',
+            id: 'anaconda-username',
+            placeholder: 'username'
+        }).on('input', function() {
+            credentials['username'] = username.val();
+        });
+
+        password = $('<input/>', {
+            'class': 'form-control',
+            type: 'password',
+            id: 'anaconda-password'
+        }).on('input', function() {
+            credentials['password'] = password.val();
+        });
+
+        $('<div/>', {'class': 'form-group'}).append(
+            $('<label/>', {'for': 'anaconda-username'})
+                .text('Username'),
+            $('<div/>').append(username))
+        .appendTo(form);
+
+        $('<div/>', {'class': 'form-group'}).append(
+            $('<label/>', {'for': 'anaconda-password'})
+                .text('Password'),
+            $('<div/>').append(password))
+        .appendTo(form);
+
+        modal = dialog.modal({
             body: body,
-            buttons : {
-                OK: {}
+            buttons: {
+                'OK': {class: 'btn-primary', click: loginIntoAnaconda},
+                'Close': {}
             }
         });
+
+        modal.find('.modal-title').append(
+            $('<span/>').text('Login into '),
```

Comment:
Perhaps the verbiage from AS should be used: _Sign In to [Anaconda Cloud]_?

Also, if we are providing login, we should probably also provide the option to set the login URL. this is of course tricky, as the user would only know the HTML URL, not the API URL. Presumably, this would just mean throwing `/api` onto anything other than `anaconda.org`